istioctl/pkg/proxystatus/proxystatus_test.go

```diff
@@ -46,8 +46,8 @@ type execTestCase struct {
 func TestProxyStatus(t *testing.T) {
 	cases := []execTestCase{
 		{ // case 0
-			args:           []string{},
-			expectedString: "NAME     CLUSTER     CDS     LDS     EDS     RDS     ECDS     ISTIOD",
+			args:          []string{},
+			wantException: true, // Error: no running Istio pods in "istio-system"
```

Comment:
seems we are removing valid test coverage here?

Reply:
The test is incorrect because the fake client returns an empty response with no error. In a real cluster, errors are shown for both commands when there is no Istiod pod running.
```
➜  istio git:(master)  istioctl ps                                 
Error: unable to find any Istiod instances
➜  istio git:(master) istioctl x ps
Error: no running Istio pods in "istio-system"
```

Reply:
every test case now gives errors, can we have a test that covers success?

Reply:
Added, thanks!